shell_sort/shellSort.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Worst case time complexity = O(n^2)
+ * Best case complexity = O(nlog(n))
+ * @param {Array} data
+ * returns sorted data
+ */
+function shellSort (data) {
+	for (let i = Math.floor(data.length / 2); i > 0; i = Math.floor(i / 2)) {
+		for (let j = i; j < data.length; j++) {
+			for (let k = j - i; k >= 0; k -= i) {
+				if (data[k + i] >= data[k]) {
+					break;
+				} else {
+					var temp = data[k];
```

Comment:
Declaring `var temp` at the top will be more efficient since it will avoid re-allocations. 
Also use `let`.